xbmc/dialogs/GUIDialogMediaSource.cpp

```diff
@@ -473,7 +473,8 @@ void CGUIDialogMediaSource::UpdateButtons()
     return;
 
   CONTROL_ENABLE_ON_CONDITION(CONTROL_OK, !m_paths->Get(0)->GetPath().empty() && !m_name.empty());
-  CONTROL_ENABLE_ON_CONDITION(CONTROL_PATH_ADD, !m_paths->Get(0)->GetPath().empty());
+  CONTROL_ENABLE_ON_CONDITION(CONTROL_PATH_ADD, !m_paths->Get(0)->GetPath().empty() &&
+                                                    !(m_paths->Size() >= 1 && m_type == "files"));
```

Comment:
just code comment, in case the PR is good to go,
you can avoid count the size
```suggestion
  CONTROL_ENABLE_ON_CONDITION(CONTROL_PATH_ADD, !m_paths->Get(0)->GetPath().empty() &&
                                                    !m_paths->IsEmpty() && m_type == "files"));
```

Reply:
Good point, thanks.